src/sentry/issues/issue_velocity.py

```diff
@@ -133,47 +134,79 @@ def calculate_threshold(project: Project) -> Optional[float]:
+
+def update_threshold(
+    project: Project,
+    threshold_key: str,
+    stale_date_key: str,
+    stale_threshold: Optional[float] = None,
+) -> float:
     """
     Runs the calculation for the threshold and saves it and the date it is last updated to Redis.
-    If the threshold is NaN, we save it as 0 at the normal TTL. If the threshold is None (due to errors
-    in the query or any other reason), we save it as 0 at the shortened TTL.
+    If the threshold is NaN, we save it as 0 at the normal TTL. If the threshold is `None` (due to
+    Snuba errors), we call the fallback method.
     """
     threshold = calculate_threshold(project)
-    ttl = DEFAULT_TTL
     if threshold is None:
-        logger.error(
-            "Velocity threshold couldn't be calculated, query returned nothing",
-            extra={"project_id": project.id},
-        )
-        threshold = 0
-        ttl = NONE_TTL
-    elif math.isnan(threshold):  # indicates there were no valid events to base the calculation
+        return fallback_to_stale_or_zero(threshold_key, stale_date_key, stale_threshold)
+    if math.isnan(threshold):
         threshold = 0
 
+    ttl = DEFAULT_TTL
     client = get_redis_client()
     with client.pipeline() as p:
         p.set(threshold_key, threshold, ex=ttl)
-        p.set(stale_date_key, convert_date_to_int(datetime.utcnow()), ex=ttl),
+        p.set(stale_date_key, str(datetime.utcnow()), ex=ttl)
         p.execute()
-
+    metrics.incr("issues.update_new_escalation_threshold", tags={"useFallback": False})
     return threshold
 
 
+def fallback_to_stale_or_zero(
+    threshold_key: str, stale_date_key: str, stale_threshold: Optional[float]
+) -> float:
+    """
+    Returns the backup threshold for when the current threshold can't be calculated. If we have a
+    stale threshold, its stale date in Redis is extended to make it usable for the next ten minutes,
+    while TTL is maintained. Otherwise, we save a value of 0 with a stale date and TTL of ten minutes
+    into the future, and return 0 (so issues in this project do not escalate during this time).
+    """
+    ttl = FALLBACK_TTL
+    # current datetime - the amount of time a threshold is valid for + how much time to wait before trying to query Snuba for the threshold again
+    stale_date = (
+        datetime.utcnow()
+        - timedelta(seconds=TIME_TO_USE_EXISTING_THRESHOLD)
+        + timedelta(seconds=FALLBACK_TTL)
+    )
+    client = get_redis_client()
+    with client.pipeline() as p:
+        if stale_threshold is None:
+            stale_threshold = 0
+            p.set(threshold_key, stale_threshold, ex=ttl)
+        else:
+            # use the stale threshold and maintain its ttl so threshold and stale date expire in redis at the same time
+            p.watch(threshold_key)
+            existing_ttl = p.ttl(threshold_key)
+            if isinstance(existing_ttl, int):
```

Comment:
```suggestion
            if isinstance(existing_ttl, int) and existing_ttl>0 :
```

Because:

- `>0` — TTL in seconds.
- `-1` — if the key exists but has no associated expiration.
- `-2` — if the key does not exist.

Reply:
ooh good callout, thanks!!